FoundryVTTRuleExtension/__init__.py

```diff
@@ -0,0 +1,87 @@
+import DatenbankEditor
+import logging
+from .DatenbankEditRegelWrapperPlus import DatenbankEditRegelWrapperPlus
+from .RegelExtension import extend_regel
+
+__version__ = "1.0.0"  # Plugin Version
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("FoundryVTTRuleExtension")
+print("FoundryVTTRuleExtension Plugin module loaded")
+
+# Categories that should show the modifications field
+FOUNDRY_CATEGORIES = [
+    "Nahkampfmanöver",
+    "Fernkampfmanöver",
+    "Magische Modifikationen",
+    "Karmale Modifikationen",
+    "Dämonische Modifikationen"
+]
+
+class Plugin:
+    def __init__(self):
+        logger.info("FoundryVTTRuleExtension Plugin initializing...")
+        
+        # Store event bus registrations
+        self.datenbank_laden_reg = EventBus.addAction("datenbank_laden", self.datenbankLadenHook)
+        self.regel_reg = EventBus.addFilter("dbe_class_regel_wrapper", self.dbeClassRegelFilter)
+
+        # Extend the Regel class with XML serialization support
+        extend_regel()
+
+    def changesCharacter(self):
+        if not hasattr(self, "db") or self.db is None or not hasattr(self.db, "einstellungen"):
+            return True
+        return self.db.einstellungen["FoundryVTTRuleExtension Plugin: Aktivieren"].wert
+
+    def changesDatabase(self):
+        return True
+
+    def datenbankLadenHook(self, params):
+        try:
+            # Initialize settings when database is loaded
+            self.db = params["datenbank"]
+            
+            # Plugin activation setting
+            e = DatenbankEinstellung()
+            e.name = "FoundryVTTRuleExtension Plugin: Aktivieren"
+            e.beschreibung = "Hiermit kannst du das FoundryVTTRuleExtension-Plugin nur für diese Hausregeln deaktivieren."
+            e.text = "True"
+            e.typ = "Bool"
+            e.kategorie = "Plugins"
+            self.db.loadElement(e)
+            
+            # Default text modification setting
+            e = DatenbankEinstellung()
+            e.name = "FoundryVTTRuleExtension Plugin: Standard Text"
+            e.beschreibung = "Standardtext für neue Regeln in Foundry VTT."
+            e.text = "Keine Modifikationen"
+            e.typ = "Text"
+            e.kategorie = "Plugins"
+            self.db.loadElement(e)
+            
+            logger.info("Settings loaded successfully")
+            
+        except Exception as e:
+            logger.error("Error in datenbankLadenHook: %s", str(e), exc_info=True)
+            raise
+
+    def dbeClassRegelFilter(self, editorType, params):
+        if not hasattr(self, "db"):
+            return editorType
+        if self.db is None:
+            return editorType
+        if not hasattr(self.db, "einstellungen"):
+            return editorType
+        if not self.db.einstellungen["FoundryVTTRuleExtension Plugin: Aktivieren"].wert:
+            return editorType
+
+        return DatenbankEditRegelWrapperPlus 
```

Comment:
Ist zwar sicherlich zumindest ungewöhnlich, aber ich lasse bei meinen plugins von editorType erben in einer Klasse, die innerhalb des Filters definiert ist. Das hat den Vorteil, dass mehrere Plugins den Editor verändern können und so in der Regel automatisch kompatibel miteinander sind. Wäre für mich aber kein Problem, es zu lassen, wie es jetzt ist. Ich würde dann aber vermutlich die Priorität beim Eventbus anpassen, dass dieser Filter vor allen anderen aufgerufen wird, somit könnten andere Plugins dann deinen DatenbankEditRegelWrapperPlus beerben.

Allgemein wäre es sicher schöner, wenn ich für jeden Editor einen Hook bereitstellen würde um etwas ins Layout einzufügen, statt die Klasse zu ändern.